### PR TITLE
[#125830841] Create bucket and enable ELB access log shipping for CF API and router

### PR DESCRIPTION
**Pull request by keymon**
What
====

We want to enable access log shipping in the ELBs to be able to troubleshoot and track any access. The logs will be ship to a S3 bucket and stored for up to 30 days (same as logsearch).

Out of scope: ingest the logs in logsearch.

Dependencies
----------

https://github.gds/government-paas/aws-account-wide-terraform/pull/52 must be merged and applied first to allow the deployer to create the bucket.

Once merged, remove the WIP commit that workarounds the lack of permissions.

How to review?
------------

 1. Check the commits and review that make sense.
 2. Deploy up to the cf-terraform job.
 3. Check that the new bucket is created. You can wait some minutes and check that there are logs being generated for each ELB.
 4. Check that the expiration lifecycle is defined, by downloading the policy with aws cli:

    ```
 aws s3api  get-bucket-lifecycle --bucket ${DEPLOY_ENV}-elb-access-log # Or ${DEPLOY_ENV}-elb-access-log-cf-droplets
```

Who?
----

Anyone but @keymon

### CODE REVIEW

File: terraform/cloudfoundry/elb_access_log_bucket.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+variable "elb_access_log_bucket_name" {
+#  default     = "elb-access-log"
+  default     = "elb-access-log-cf-droplets"
+}
```
Comment (mtekel) [inline terraform/cloudfoundry/elb_access_log_bucket.tf]: why cf-droplets?

Comment (keymon) [inline terraform/cloudfoundry/elb_access_log_bucket.tf]: Why not?

I am bypassing the policy. We are allowed to create `*-cf-droplets` among others. That should be deleted once the PR on the polices is merged.

Comment (mtekel) [inline terraform/cloudfoundry/elb_access_log_bucket.tf]: ok, didn't notice it was in a temp/wip commit

**Comment by mtekel** [comment]
works, will re-test after IAM policy is merged & remove WIP & merge then

**Comment by alext** [comment]
https://github.gds/government-paas/aws-account-wide-terraform/pull/52 has now been merged and applied to all environments.

**Comment by mtekel** [comment]
I removed the WIP commit & it still works. There is one catch though. TF keeps flapping and it updates bucket policy on every run:
```
aws_s3_bucket.elb_access_log: Modifying...
  policy: "{\"Statement\":[{\"Action\":\"s3:PutObject\",\"Effect\":\"Allow\",\"Principal\":{\"AWS\":\"arn:aws:iam::156460612806:root\"},\"Resource\":\"arn:aws:s3:::mt-elb-access-log/*/AWSLogs/*/*\"}],\"Version\":\"2012-10-17\"}" => "{\"Statement\":[{\"Action\":[\"s3:PutObject\"],\"Effect\":\"Allow\",\"Principal\":{\"AWS\":[\"156460612806\"]},\"Resource\":[\"arn:aws:s3:::mt-elb-access-log/*/AWSLogs/*/*\"]}],\"Version\":\"2012-10-17\"}"
```

**Comment by keymon** [comment]
Good catch @mtekel. I added a commit to workaround that, please have a look.

**Comment by mtekel** [comment]
thanks, merging

**Pull request merged**